### PR TITLE
Contribution docs

### PR DESCRIPTION
**Pull request by bdeanindy**
This change will add the formal Contributor License Agreement (CLA) as well as documenting the Pull Request process for other contributions later.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Contributing to SparkPost
+
+Transparency is one of our core values, and we encourage developers to contribute and become part of the SparkPost developer community.
+## Prerequisite to contribution
+Part of this transparency is the perpetual management. Every developer (even our own) to get started is required, <a href="https://www.clahub.com/agreements/SparkPost/node-sparkpost">sign the Contributor License Agreement</a>. Your pull requests will be flagged as "has not signed CLA" unless you agree to the CLA.
```
Comment (bdeanindy) [inline CONTRIBUTING.md]: We're just going to remove the entire CLA (after our discussion).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+1. Fork this repository
+2. Create a new branch named after the issue you’ll be fixing (include the issue number as the feature branch name)
+3. Read and apply our [code style guide](CODE_STYLE_GUIDE.markdown). Make sure your submission meets our contribution guidelines.
+4. Author tests, code to satisfy, clean up/refactor
+    1. Include your tests in the 'test' directory in the appropriate test file
+    2. To test the SDK
+    ```
+    grunt test
+    ```
+5. Submit a new Pull Request applying your feature/fix branch to the develop branch of the SparkPost SDK
```
Comment (bdeanindy) [inline CONTRIBUTING.md]: Let me create an issue for that (and that will be the first thing that I fix)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Part of this transparency is the perpetual management. Every developer (even our own) to get started is required, <a href="https://www.clahub.com/agreements/SparkPost/node-sparkpost">sign the Contributor License Agreement</a>. Your pull requests will be flagged as "has not signed CLA" unless you agree to the CLA.
+
+Before writing code search for issues or create a new issue to confirm where your contribution fits into our roadmap.
+Current milestone Pull Requests will receive priority review for merging.
+
+## Contribution Steps
+1. Fork this repository
+2. Create a new branch named after the issue you’ll be fixing (include the issue number as the feature branch name)
+3. Read and apply our [code style guide](CODE_STYLE_GUIDE.markdown). Make sure your submission meets our contribution guidelines.
+4. Author tests, code to satisfy, clean up/refactor
```
Comment (bdeanindy) [inline CONTRIBUTING.md]: I will get this changed now.

File: README.md [hunk 4]
```diff
@@ -95,10 +95,10 @@ Run `npm install` inside the repository to install all the dev dependencies.
 Once all the dependencies are installed, you can execute the unit tests using `grunt test`
 
 ### Contributing
-Guidelines for adding issues
+To get started, <a href="https://www.clahub.com/agreements/SparkPost/node-sparkpost">sign the Contributor License Agreement</a>.
```
Comment (bdeanindy) [inline README.md]: Nuff said, I'll pull this out as well.

**Comment by bdeanindy** [comment]
Our Code Style Guide will be different for each language (so until we have multiple repositories in the same language, let's keep it close to the code...makes it easier to find for devs).

We're removing the CLA (I'll patch that and resubmit the PR in just a few).

**Comment by bdeanindy** [comment]
FYI, I've removed the CLA hook that was in place.

**Comment by bdeanindy** [comment]
@richleland Could you take a look at the latest revs please sir?

**Comment by richleland** [comment]
I think this if fine. Everything feels a little verbose to me to be honest, but we can just do subsequent PRs for any tweaks to language and not hold this up. @bizob2828 if you agree let's merge.

**Comment by bizob2828** [comment]
:shipit:

**Comment by bizob2828** [comment]
merging now

**Pull request merged**